### PR TITLE
documentation: the SM model parallel library v1.15.0 release note

### PR DESCRIPTION
**Pull request by mchoi8739**
*Description of changes:*

the SM model parallel library v1.15.0 release note

*Testing done:*

The DistributedTransformerLayer class: 
https://sagemaker--3806.org.readthedocs.build/en/3806/api/training/smp_versions/latest/smd_model_parallel_pytorch_tensor_parallel.html#smdistributed.modelparallel.torch.nn.DistributedTransformerLayer

Release note:
https://sagemaker--3806.org.readthedocs.build/en/3806/api/training/smd_model_parallel_release_notes/smd_model_parallel_change_log.html

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by knikure — APPROVED** [review]

File: doc/api/training/smd_model_parallel_release_notes/smd_model_parallel_change_log.rst [hunk 1]
```diff
@@ -3,12 +3,88 @@ Release Notes
...
-SageMaker Distributed Model Parallel 1.14.0 Release Notes
+SageMaker Distributed Model Parallel 1.15.0 Release Notes
 =========================================================
 
+*Date: Apr. 25. 2023*
+
+**Currency Updates**
+
+* Added support for PyTorch v2.0.0.
+  However, the library does not support ``torch.compile`` at this release.
```
Comment (liuxl89) [inline doc/api/training/smd_model_parallel_release_notes/smd_model_parallel_change_log.rst:17]: LGTM.

Just a quick comment, not sure which one is better, e.g.
- However, the library does not support ``torch.compile`` at this release. vs
- Note that the library does not support ``torch.compile`` at this release yet.

Comment (mchoi8739) [reply doc/api/training/smd_model_parallel_release_notes/smd_model_parallel_change_log.rst]: updated the sentence to incorporate with your second suggestion.

**Review by viclzhu — APPROVED** [review]
Everything LGTM!

**Review by liuxl89 — APPROVED** [review]

**Review by knikure — COMMENTED** [review]
/bot run unit-tests

**Pull request merged**